Formula/rm-improved.rb

```diff
@@ -0,0 +1,23 @@
+class RmImproved < Formula
+  desc "Command-line deletion tool focused on safety, ergonomics, and performance"
+  homepage "https://github.com/nivekuil/rip"
+  url "https://github.com/nivekuil/rip/archive/0.13.1.tar.gz"
+  sha256 "73acdc72386242dced117afae43429b6870aa176e8cc81e11350e0aaa95e6421"
+  license "GPL-3.0-or-later"
+  head "https://github.com/nivekuil/rip.git"
+
+  livecheck do
+    url "https://github.com/nivekuil/rip/releases/latest"
+    regex(%r{href=.*?/tag/v?(\d+(?:\.\d+)+)["' >]}i)
+  end
+
+  depends_on "rust" => :build
+
+  def install
+    system "cargo", "install", *std_cargo_args
+  end
+
+  test do
+    system "false"
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
Thanks for the advice. I have an idea for a test, its just a matter of figuring it out in Ruby + running the test assertions.

I am just rubber ducking this in bash for myself:

```bash
# setup test
touch hello-world

rip hello-world
# assert file is in /tmp/graveyard-$USER

rip --unbury
# assert file is back where it was deleted from
```

If using the `/tmp` path is problematic, I can set the graveyard to something relative to the test path (e.g. `rip --graveyard .graveyard`).

Reply:
Tests added in 43dc99d